### DISCUSSION TITLE
RS485-Verbindung Fusion Board / US2000C

### DISCUSSION THREAD

**Discussion by huey84** [hoylabs/OpenDTU-OnBattery#1168] [Q&A] [answered]
Hallo in die Runde, 

ausgelöst durch die Unterhaltung #944 versuche ich eine RS485-Verbindung zwischen einem Fusion Board v2.1 (plus POE-  & CAN/ISO-Shield) und einer Pylontech US2000C zum Laufen zu bringen, was mir nicht gelingt. Eine Verbindung via CAN-Bus klappt sofort und fehlerfrei, d.h. die Komponenten funktionieren. 

Hat jemand einen Tipp für mich?

**Bisherige Erkenntnisse:**
- Verwendete FW: 2024.06.03
- Mapping:  "battery": { "rx": 16, "rxen": 15, "tx": 45, "txen": 46 } wie in #944 angegeben.
- HW-Einstellungen: Die Pins aus dem Mapping (16,15,45,46) werden auch als „Aktiv“ angezeigt.
- Batterie-Einstellungen: Batterie ist aktiviert, Ausführliches Protokoll an, „BMS per serieller Verbindung / RS485 Transceiver an der MCU“ ausgewählt, Intervall ist 3s
- Verwendet wird ein 1:1 RJ-45 Kabel, nur Pin 7 und 8. Gesteckt auf US2000C PortA/CAN (oder an PortB/RS485) und Fusion POE Shield RS485 Port A und B
- RS485 Baudrate an US2000C auf 115000

**Es kommt keine Verbindung zustande. Die Konsole sagt alle 6sec dieses:**

> 12:43:40.716 > [   1025.592] JK BMS: timeout wating for response from BMS
> 12:43:40.734 > [   1025.598] JK BMS: request for data sent
> 12:43:40.738 > [   1025.601] JK BMS: busy waiting for or reading a message from the BMS

**Was wurde bisher alles mit dem gleichen, erfolglosen Ergebnis ausprobiert:**
- Alternativ das „blaue“ Pylontech CAN-Kabel benutzt
- A/B vertauscht
- Zusätzlich mit 120 Ohm am CAN/ISO Shield terminiert (mit 1:1 und A/B getauscht getestet)
- Baudrate auf 9600 (DIP 1 an US2000C auf ON, dann Reboot)
- Ohne POE- & CAN/ISO-Shield, also nur mit Fusion Board getestet

**Comment by AndreasBoehm** [comment, marked as answer]
Es gibt derzeit keinen support für Pylontech bzw Pytes Batterien via RS485, geht nur per CAN.

Die Option die du gefunden hast ist für externe BMS von DIY Batterien.

**Reply by huey84** [reply]
Ok, danke für das Feedback. Dann hatte ich da wohl was falsch verstanden oder war einfach zu ungeduldig. In #622 sah es so aus, als sei die Unterstützung für RS485 / Pylontech schon sehr weit fortgeschritten. 

Dann warte ich geduldig, bis der Support ins nächste (?) Release einfließt ;o)

**Comment by Manos1966** [comment]
OpenDTU-OnBattery unterstuzt nur das, was im Wiki aufgelistet ist:
https://github.com/helgeerbe/OpenDTU-OnBattery/wiki#highlights-of-opendtu-onbattery

Falls du unbedingt RS486 haben willst, es existiert zur Zeit ein Fork von der urspruenglichen OpenDTU-OnBattery:
https://github.com/skippermeister/OpenDTU-OnBattery

**Reply by huey84** [reply]
Vielen Dank für den Hinweis & Tipp. Ich werde es mir anschauen.

**Reply by Manos1966** [reply]
Ich nutze meine "normale" 🙈 OpenDTU-OnBattery fuer die algemeine Steuerung.
Parallel dazu, habe ich meine zwei Batterien mit Skippermeister's OpenDTU-OnBattery version ueber RS485 angeschlossen.

**Reply by huey84** [reply]
Interessant! Und die beiden DTUen unterhalten sich wie? Per MQTT?

**Reply by Manos1966** [reply]
gar nicht. 
Die zweite DTU liesst nur die Daten der zwei Pylontech ueber RS485 und gibt mir die detaillierte Infos ueber jede der 15 Zellen.
![Master](https://github.com/user-attachments/assets/e2431ed1-3daf-4d0c-ad4f-6e6fafe8fc08)

**Reply by huey84** [reply]
Verstanden. Die Steuerung der Batterien via CAN mit der „Standard“ DTU1, die Überwachung per RS485 mit „Special“-DTU2. Damit wäre ich mit ein paar Euros mehr am Ziel. Prima, danke für die  Idee!

**Reply by huey84** [reply]
Welche Baudrate bei RS485? 115000? Oder doch nur 9600? (DIP Schalter 1 an US2000)

**Reply by Manos1966** [reply]
ich habe zwei US5000, ich habe die DIP Schalter nicht angefasst...

**Reply by skippermeister** [reply]
@huey84 
Baudrate auf 115200.
